### PR TITLE
Add how-to guide for testing tail sampling policies

### LINKED ISSUE #39: How-to guide: testing tail sampling policies
## Context

Tail sampling is the 3rd most common pain point in the OpenTelemetry community (167 Slack messages analysed in the [2026 community insights post](https://opentelemetry.io/blog/2026/slack-community-insights/)). Users struggle with policy configuration, multi-policy interactions, and debugging sampling decisions.

motel can generate deterministic, repeatable trace traffic with known characteristics (error rates, durations, attribute distributions), making it ideal for testing tail sampling policies without deploying real services.

## What to write

A how-to guide (`docs/how-to/test-tail-sampling.md`) covering:

1. A simple topology that produces traces with a mix of characteristics (errors, slow spans, specific attributes) that exercise common sampling policies
2. Example collector config with tail sampling processor
3. Walkthrough: run motel → collector with tail sampling → verify what gets sampled
4. How to adjust the topology to test edge cases (e.g. "what happens when 100% of traces are slow?")

## Example topology to include

An example topology file in `docs/examples/` that generates traces specifically useful for sampling testing — a mix of:
- Normal fast traces (majority)
- Slow traces (high duration)
- Error traces
- Traces with specific attribute patterns (e.g. VIP customer flag)

This gives users known input to verify their sampling policies produce expected output.

## Why this matters

Today, testing tail sampling requires either live traffic or hand-crafted test harnesses. motel already has the building blocks — this guide shows users how to connect them.

### PR DESCRIPTION
**Pull request by andrewh**
## Summary
- Add `docs/how-to/test-tail-sampling.md` -- walkthrough for using motel to test tail sampling policies in an OpenTelemetry Collector
- Add `docs/examples/tail-sampling-test.yaml` -- example topology producing a mix of normal, slow, error, and VIP traces
- Covers: topology design, collector config with tail sampling processor, running motel against the collector, verifying sampling results, and adjusting for edge cases

## Test plan
- [x] `motel validate` passes on the example topology
- [x] `motel run --stdout` generates traces with expected span names
- [ ] Links in the guide resolve correctly on GitHub

Closes #39

### CODE REVIEW

File: docs/how-to/test-tail-sampling.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+This verifies that your probabilistic baseline still applies when the attribute-based policy matches most traces.
+
+### Test with scenarios labelled
+
+Use the `--label-scenarios` flag to add `synth.scenario` attributes to spans, so you can see which scenario was active when a trace was generated:
+
+```sh
+motel run --stdout --duration 15s --label-scenarios \
+  docs/examples/tail-sampling-test.yaml | \
+  jq -r '.Attributes["synth.scenario"] // "baseline"' | \
```
[the remark this answers is not in the document]
Comment (andrewh) [reply docs/how-to/test-tail-sampling.md]: Agreed. Added root span filter (using the correct field path — .Parent.SpanID, not .ParentSpanID).

**Comment by andrewh** [comment]
Consolidated into PR 60
